### DISCUSSION TITLE
OAuth: Allow redirect url

### DISCUSSION THREAD

**Discussion by eXpire163** [grafana/grafana#42237] [General]
In some enterprise context oAuth requires some custom redirect logic, allowing to overwrite the default redirect url for the oAuth process would enable this.

I understand that this is an edge case, but it would allow us to run our 140 grafana stacks against a single github oauth app, that then redirects to our distribution nginx (trampoline in the diagram), forwarding back to the dedicated domain of each grafana.

![image](https://user-images.githubusercontent.com/16032218/143259007-be986af2-46e3-4bfd-bce4-c8ca26eff488.png)

PR: https://github.com/grafana/grafana/pull/41914

**Comment by marefr** [comment]
Yes, this is an edge case that probably not many others have problem with/in need of. 

To me sounds like [Auth proxy](https://grafana.com/docs/grafana/latest/auth/auth-proxy/) or [JWT authentication](https://grafana.com/docs/grafana/latest/auth/jwt/) would suit you better to give you more control of the whole authentication flow. Have you considered any of these?

I'm going to close the pull request for now since we're trying to only keep pull requests open that we've accepted as changes we want. Lets continue the discussions in here to see if there's any other interest for this. A closed PR can always be opened again.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
